torch/utils/data/dataloader.py

```diff
@@ -116,15 +125,23 @@ class DataLoader(object):
               details on these two types of datasets and how
               :class:`~torch.utils.data.IterableDataset` interacts with `Multi-process data loading`_.
     """
+    dataset: Dataset[T_co]
+    batch_size: Optional[int]
+    num_workers: int
+    pin_memory: bool
+    drop_last: bool
+    timeout: float
```

Comment:
Is it possible to list `sampler: Sampler` here? Some internal code attempts to access `dataloader.sampler` and the type appears was inferred to be Optional[Sampler]. However, after `__init__`, we're sure that sampler is a Sampler

Reply:
(I verified that adding `sampler: Sampler` makes the problem go away, but I'm not sure if there was a reason why it wasn't here)